File: DSAlgorithms/Leetcode/c++/arrays/arrays_hash/217-contains_duplicates.cpp
class Solution {
public:
    bool containsDuplicate(vector<int>& nums) {
        std::unordered_map<int, int> hash;
        for(int i = 0; i < nums.size(); i++)
        {
            if(hash.count(nums[i]) > 0)
                return true;
            hash[nums[i]] = 1;
        }
        return false;
    }
};

File: DSAlgorithms/Leetcode/ts/src/arrays_hash/217-contains_duplicates.ts
const contains_duplicates = (nums:number[]): boolean => {
    const hash = new Map<number, number>();
    for(let i = 0; i < nums.length; i++)
    {
        if(hash.get(nums[i]))
            return true;
        hash.set(nums[i], 1);
    }
    return false;
};

let arr: number[] = [1, 2, 3, 2, 4, 5];

console.log(contains_duplicates(arr));

